benchmarks/opmphm.c

```diff
@@ -980,6 +989,787 @@ static void benchmarkMappingAllSeeds (void)
  * END ==================================================== Mapping All Seeds ========================================================== END
  */
 
+/**
+ * START ================================================== OPMPHM Build Time ======================================================== START
+ *
+ * This benchmark measures the time of the OPMPHM build.
+ * Uses all KeySet shapes, for all n (KeySet size) a fixed set of seeds is used to build the OPMPHM.
+ * For one n (KeySet size) ksPerN KeySets are used.
+ * The results are written out in the following format:
+ *
+ * n;ks;time
+ *
+ * The number of needed seeds for this benchmarks is: numberOfShapes * ( numberOfSeeds + nCount * ksPerN )
+ */
+
+/**
+ * @brief Measures the OPMPHM build numberOfRepeats time and returns median
+ *
+ * @param ks the KeySet
+ * @param repeats array to store repeated measurements
+ * @param numberOfRepeats fields in repeats
+ *
+ * @retval median time
+ */
+static size_t benchmarkOPMPHMBuildTimeMeasure (KeySet * ks, size_t * repeats, size_t numberOfRepeats)
+{
+	for (size_t repeatsI = 0; repeatsI < numberOfRepeats; ++repeatsI)
+	{
+		// preparation for measurement
+		struct timeval start;
+		struct timeval end;
+		Key * keySearchFor = ks->array[0]; // just some key
+		Key * keyFound;
+		// fresh OPMPHM
+		if (ks->opmphm)
+		{
+			opmphmDel (ks->opmphm);
+			ks->opmphm = NULL;
+		}
+
+		// START MEASUREMENT
+		__asm__("");
+		gettimeofday (&start, 0);
+		__asm__("");
+
+		keyFound = ksLookup (ks, keySearchFor, KDB_O_OPMPHM | KDB_O_NOCASCADING);
```

Comment:
Isn't a single lookup too fast to be measured precisely?

Reply:
This single lookup builds the OPMPHM, thus it takes some time.